addon/lib/user.js

```diff
@@ -64,3 +68,129 @@ function makeUuid() {
 exports.getUserInfo = function () {
   return ss.storage.userInfo;
 };
+
+exports.updateProfileInfo = function (profile) {
+  if (! profile) {
+    return;
+  }
+  let currentInfo = ss.storage.profileInfo;
+  if (! currentInfo) {
+    currentInfo = ss.storage.profileInfo = {};
+  }
+  let attrs = Object.keys(profile);
+  for (let attr of attrs) {
+    if (! currentInfo[attr] || profile[attr] && currentInfo[attr] != profile[attr]) {
+      currentInfo[attr] = profile[attr];
+    }
+  }
+  ss.storage.profileInfo = currentInfo;
+};
+
+exports.getProfileInfo = function () {
+  return ss.storage.profileInfo;
+};
+
+exports.updateLogin = function (backend, info) {
```

Comment:
I think this more means `updateProfile` or `updateAccount`?  This is basically updating the `accounts` table with profile info for the user, right?  I suppose it is also assigning the reference between the device and the account...?